### PR TITLE
Fix : As a dev, I want to fix the swagger on the deployment #136

### LINKED ISSUE #136: As a dev, I want to fix the swagger on the deployment
## Description  

**Bug Summary:** 
On the deployed `fertiscan-backend`, the swagger is not functional

**Steps to Reproduce:**  
https://fertiscan.inspection.alpha.canada.ca/api/apidocs

**Expected Behavior:**  
Should display the swagger doc.

**Actual Behavior:**  
Doesn't load the swagger.

**Suggested Fix:**
Check the root path where swagger is mounted.

### PR DESCRIPTION
**Pull request by Youllou**
Fix #136 

What has been done :
- added a new env value API_BASE_PATH
- added swagger_config to be able to modify the config ( it is not modifiable once flasgger has been created )
- modify the config to change the url_prefix by the env value (changes the url for swagger ui)
- modify the template to change the basePath by the env value (changes the url for the api routes)

image

### CODE REVIEW

**Comment by Youllou** [comment]
Note that the env value should be added to the vault !

**Review by k-allagbe — COMMENTED** [review]
Looks good. Please, 
- create a new vault secret version for this, and open a PR in howard to change both the secret version and the deployment image
- redeploy
- test

**Comment by SonOfLope** [comment]
Would it be possible to add a default value so we know whats expected to use when we run it locally with the docker-compose. The env section 'Environment Variables' in README should also be updated with new values.

**Review by ThomasCardin — APPROVED** [review]

**Comment by k-allagbe** [comment]
After trying many things, the swagger is still not served in deployment.

Might be related to https://flask.palletsprojects.com/en/3.0.x/deploying/proxy_fix/

@SonOfLope or @ThomasCardin Could you please take a look at this when you can?

**Comment by SonOfLope** [comment]
README needs to be updated to new endpoint for swagger page (/swagger/docs)

File: app.py [hunk 1]
```diff
@@ -79,9 +91,11 @@
 )
 connection_manager = ConnectionManager(app, pool)
 
-@app.route('/', methods=["GET"])
+
+@app.route("/", methods=["GET"])
 def main_page():
-    return redirect("/apidocs/", code=HTTPStatus.FOUND)
+    return redirect(BASE_PATH + "/docs")
```
Comment (SonOfLope) [inline app.py:97]:
```suggestion
    return redirect(BASE_PATH + "SWAGGER_PATH")
```

File: .env.template [hunk 2]
```diff
@@ -12,6 +12,9 @@ UPLOAD_PATH='./uploads'
 # For CORS purposes
 FRONTEND_URL=""
 
+# API base path for deployment
+API_BASE_PATH='/swagger'
+
```
Comment (SonOfLope) [inline .env.template:17]:
```suggestion
SWAGGER_PATH='/docs'
```

**Review by k-allagbe — APPROVED** [review]

**Pull request merged**